### PR TITLE
[2.4] Cherry-picks from master for v2.4.11

### PR DESCRIPTION
**Pull request by alonre24**
After reverting previous cherry-pick PR - this is the updated cherry picks for 2.4.11 (without hybrid optional arguments in vecsim)

### CODE REVIEW

**Review by ashtul — DISMISSED** [review]

File: coord/src/dist_aggregate.c [hunk 1]
```diff
@@ -401,12 +401,6 @@ void RSExecDistAggregate(RedisModuleCtx *ctx, RedisModuleString **argv, int argc
   int rc = AREQ_Compile(r, argv + 2 + profileArgs, argc - 2 - profileArgs, &status);
   if (rc != REDISMODULE_OK) goto err;
 
-  // Save time when query was initiated
-  if (!r->reqTimeout) {
-    r->reqTimeout = RSGlobalConfig.queryTimeoutMS;
-  }
-  updateTimeout(&r->timeoutTime, r->reqTimeout);
-
```
Comment (DvirDukhan) [inline coord/src/dist_aggregate.c:409]: is this need to be removed?

File: src/query_error.h [hunk 2]
```diff
@@ -44,7 +44,14 @@ extern "C" {
...
   X(QUERY_ENOTNUMERIC, "Could not convert value to a number")                   \
-  X(QUERY_ENOPARAM, "Parameter not found")
+  X(QUERY_TIMEDOUT, "Timeout limit was reached")                                \
+  X(QUERY_ENOPARAM, "Parameter not found")                                      \
+  X(QUERY_EDUPPARAM, "Parameter was specified twice")                           \
+  X(QUERY_EBADVAL, "Invalid value was given")                                   \
+  X(QUERY_ENHYBRID, "hybrid query attributes were sent for a non-hybrid query") \
+  X(QUERY_EHYBRIDNEXIST, "invalid hybrid policy was given")                     \
+  X(QUERY_EADHOCWBATCHSIZE, "'batch size' is irrelevant for 'ADHOC_BF' policy") \
+  X(QUERY_EADHOCWEFRUNTIME, "'EF_RUNTIME' is irrelevant for 'ADHOC_BF' policy") \
```
Comment (DvirDukhan) [inline src/query_error.h:54]: I think those are not related here

File: tests/pytests/test_vecsim.py [hunk 3]
```diff
@@ -1289,3 +1288,63 @@ def test_rdb_memory_limit():
...
+            vector = np.random.rand(1, dim).astype(np.float32)
+            conn.execute_command('HSET', n_vec+i, 'vector', vector.tobytes(), 't', 'dummy')
+        prefix = "_" if env.isCluster() else ""
+        for mode in hybrid_modes:
+            # to trigger ad-hoc BF, use a filter that is passed by a ~5% of the vectors, and for batches use a filter
+            # that is passed by ~95% of the vectors.
+            hybrid_query_filter = "dummy" if mode == 'HYBRID_ADHOC_BF' else '-dummy'
+            n_res = int(n_vec/20) if mode == 'HYBRID_ADHOC_BF' else n_vec
+            # res = conn.execute_command('FT.SEARCH', 'idx', '(-dummy)=>[KNN $K @vector $vec_param HYBRID_POLICY $hp]', 'NOCONTENT', 'LIMIT', 0, n_vec,
+            #                            'PARAMS', 6, 'K', n_vec, 'vec_param', query_vec.tobytes(), 'hp', mode,
```
Comment (DvirDukhan) [inline tests/pytests/test_vecsim.py:1333]: remove comment?

File: tests/pytests/test_vecsim.py [hunk 4]
```diff
@@ -1289,3 +1288,63 @@ def test_rdb_memory_limit():
...
+            #                            'TIMEOUT', 0)
+            res = conn.execute_command('FT.SEARCH', 'idx', f'({hybrid_query_filter})=>[KNN $K @vector $vec_param]', 'NOCONTENT', 'LIMIT', 0, n_res,
+                                       'PARAMS', 4, 'K', n_res, 'vec_param', query_vec.tobytes(),
+                                       'TIMEOUT', 0)
+            env.assertEqual(res[0], n_res)
+            env.assertEqual(env.cmd(prefix + "FT.DEBUG", "VECSIM_INFO", "idx", "vector")[-1], mode)
+
+            # res = conn.execute_command('FT.SEARCH', 'idx', '(-dummy)=>[KNN $K @vector $vec_param HYBRID_POLICY $hp]', 'NOCONTENT', 'LIMIT', 0, n_vec,
+            #                            'PARAMS', 6, 'K', n_vec, 'vec_param', query_vec.tobytes(), 'hp', mode,
+            #                            'TIMEOUT', 1)
```
Comment (DvirDukhan) [inline tests/pytests/test_vecsim.py:1343]: remove comment?

**Review by DvirDukhan — CHANGES_REQUESTED** [review]

File: coord/src/dist_aggregate.c (hunk 1 above)
Comment (alonre24) [reply coord/src/dist_aggregate.c:409]: This removal was part of Ariel's PR, right? @ashtul

File: src/query_error.h (hunk 2 above)
Comment (alonre24) [reply src/query_error.h:54]: right...

File: coord/src/dist_aggregate.c (hunk 1 above)
Comment (ashtul) [reply coord/src/dist_aggregate.c:409]: @DvirDukhan @alonre24 It was added to function `AREQ_Compile`.

**Review by DvirDukhan — APPROVED** [review]

**Comment by DvirDukhan** [comment]
@alonre24 what about #2864 ?

**Comment by alonre24** [comment]
> @alonre24 what about #2864 ?

That PR fixed a part of the test that was added in the "hybrid optional arguments" PR which is not in 2.4
